src/common/utility/array-replace.ts

```diff
@@ -8,6 +8,7 @@ export interface ArrayReplaceOptions {
     matchObject(objectA: any, objectB: any): boolean;
 }
 
+/* eslint-disable import/export */
```

Comment:
I'm assuming the issue here is with the multiple `export default`, can we define the function multiple times and export once?

```ts
function arrayReplace<T>(currentArray: T[] | undefined, newArray: T[], options?: ArrayReplaceOptions): T[];
function arrayReplace<T>(currentArray: T[], newArray?: T[], options?: ArrayReplaceOptions): undefined;
function arrayReplace<T>(currentArray?: T[], newArray?: T[], options?: ArrayReplaceOptions): T[] | undefined;

export default arrayReplace;
```